src/template/render.rs

```diff
@@ -471,9 +473,45 @@ impl<'a> ChainTemplateSource<'a> {
         };
         debug_span!("Executing command", ?command)
             .in_scope(|| async {
-                let output = Command::new(program)
+                // Spawn the command process
+                let mut process = Command::new(program)
                     .args(args)
-                    .output()
+                    .stdin(Stdio::piped())
+                    .stdout(Stdio::piped())
+                    .stderr(Stdio::piped())
+                    .spawn()
+                    .map_err(|error| ChainError::Command {
+                        command: command.to_owned(),
+                        error,
+                    })
+                    .traced()?;
+
+                // Write to stdin if present
+                if let Some(template) = stdin {
+                    let rendered = template
+                        .render_stitched(context)
+                        .await
+                        .map_err(|error| ChainError::Nested {
+                            field: "stdin".into(),
+                            error: error.into(),
+                        })?;
+
+                    process
+                        .stdin
+                        .take()
+                        .unwrap()
```

Comment:
I'm guessing you got this from [the docs](https://docs.rs/tokio/latest/tokio/process/struct.Child.html#structfield.stdin), but in this case we don't need `take()` because we're done with stdin before we do anything else with the process.

```suggestion
                    process
                        .stdin
                        .get_mut()
                        .expect("Process missing stdin")
```